src/main/kotlin/me/fornever/todosaurus/ui/wizard/TodosaurusWizard.kt

```diff
@@ -0,0 +1,356 @@
+            nextStep.previousId = dynamicStep.id
+        }
+
+        addStep(dynamicStep)
+
+        myCurrentStep = mySteps.size - 1
+        dynamicSteps.put(myCurrentStep, dynamicStep)
+
+        updateStep(SwipeDirection.FORWARD)
+    }
+
+    override fun doOKAction() {
+        if (!this.okAction.isEnabled)
+            return
+
+        applyFields()
+
+        if (rememberUserChoiceCheckBox.isSelected) {
+            mySteps
+                .filterIsInstance(MemorableStep::class.java)
+                .forEach {
+                    it.rememberUserChoice()
+                }
+        }
+
+        CoroutineScope(Dispatchers.IO).launch {
+            val result = finalAction()
+
+            if (result == WizardResult.Success) {
+				withContext(Dispatchers.EDT) {
+					close(0)
+				}
+            }
+        }
+    }
+
+    override fun getNextStep(stepIndex: Int): Int {
+        val step = indexesToSteps.get(stepIndex)
+        return stepsToIndexes.getInt(step.nextId)
+    }
+
+    override fun getPreviousStep(stepIndex: Int): Int {
+        val step = indexesToSteps.get(stepIndex)
+        return stepsToIndexes.getInt(step.previousId)
+    }
+
+    override fun updateStep() {
+        super.updateStep()
+        updateButtons()
+
+        currentStepObject
+            .preferredFocusedComponent
+            ?.let {
+                IdeFocusManager
+                    .findInstanceByComponent(window)
+                    .requestFocus(it, true)
+            }
+    }
+
+    override fun updateButtons() {
+        super.updateButtons()
+
+        val currentStep = indexesToSteps.get(myCurrentStep)
+        previousButton.isEnabled = currentStep.previousId != null
+        nextButton.isEnabled = currentStep.isComplete() && !isLastStep || isLastStep && canFinish()
+        rememberUserChoiceCheckBox.isVisible = currentStep is MemorableStep
+    }
+
+    override fun createSouthPanel(): JComponent {
+        if (useDialogWrapperSouthPanel()) {
+            return super.createSouthPanel()
+        }
+
+        val hasOnlyDynamicStepProvider = mySteps.size == 1 && mySteps[0] is DynamicStepProvider
+
+        val verticalPanel = JPanel().also {
+            it.layout = BoxLayout(it, BoxLayout.Y_AXIS)
+        }
+
+        if (style == DialogStyle.COMPACT) {
+            verticalPanel.border = BorderFactory.createEmptyBorder(4, 15, 4, 15)
+        }
+
+        val topPanel = JPanel().also {
+            it.layout = BorderLayout()
+        }
+
+        val bottomPanel = JPanel()
+
+        // TODO[#38]: Uncomment this condition while implementing task #38 (This will create a “Remember my choice” checkbox for the desired steps)
+        /*if (mySteps.any { it is MemorableStep }) {
+            JPanel(BorderLayout()).also {
+                it.add(rememberUserChoiceCheckBox, BorderLayout.CENTER)
+                topPanel.add(it, BorderLayout.EAST)
+            }
+        }*/
+
+        if (SystemInfo.isMac) {
```

Comment:
Sorry, why is this so radically different for Mac and non-Mac?

Reply:
**Why did I do it here?**
I copied the code from `intellij-communit`y and rewrote the layout to add a "Remember my choice" checkbox.  This way, the design of the wizard remains unchanged.

**Why did `intellij-community` developers do this?**
This question should be addressed to `intellij-community` developers :p

**My opinion**
I think we should follow the IDE design and not change the button layout

Reply:
Please note that copying the code requires us to flow the copyright information as well.

For now, add a note with link to the original code here. I'll manage the licensing myself then. This is not a problem per se, but a thing that must be properly tracked.